aten/src/ATen/native/Pow.cpp

```diff
@@ -0,0 +1,70 @@
+#include <ATen/native/Pow.h>
+
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/native/TensorIterator.h>
+
+namespace at { namespace native {
+
+DEFINE_DISPATCH(pow_tensor_tensor_stub);
+DEFINE_DISPATCH(pow_tensor_scalar_stub);
+DEFINE_DISPATCH(pow_scalar_tensor_stub);
+
+Tensor& pow_out(Tensor& result, const Tensor& base, const Tensor& exp) {
+  auto iter = TensorIterator::binary_op(result, base, exp);
+  pow_tensor_tensor_stub(iter.device_type(), iter);
+  return result;
+}
+
+Tensor& pow_out(Tensor& result, const Tensor& base, Scalar exp) {
+  // Numpy compatibility check:
+  TORCH_CHECK(!(isIntegralType(base.scalar_type()) &&
+              exp.isIntegral() && exp.toLong() < 0),
+              "Integers to negative integer powers are not allowed.");
+  if (exp.toDouble() == 0.0) {
+    result.copy_(ones(base.sizes(), base.options()));
```

Comment:
Doing resize_as and fill_(1) will give you better performance.

Reply:
Addressed